libs/ui/search/src/lib/record-table/record-table.component.ts

```diff
@@ -32,4 +40,38 @@ export class RecordTableComponent {
     }
     return undefined
   }
+
+  getRecordFormats(record: CatalogRecord): FileFormat[] {
+    if (record.kind === 'service') {
+      return []
+    }
+    const formatsAndPrio = record.distributions.map(
+      (distribution: DatasetDistribution) => {
+        return {
+          format: getFileFormat(distribution),
+          priority: sortPriority(distribution),
+        }
+      }
+    )
+
+    const sortedFormats = formatsAndPrio
+      .sort((a, b) => a.priority - b.priority)
+      .map((element) => element.format)
+
+    const types = new Set(sortedFormats)
+    return Array.from(types).filter((elm) => elm)
+  }
+
+  getBadgeColor(format: FileFormat): string {
+    return getBadgeColor(format)
+  }
+
+  firstFormat(record: CatalogRecord): FileFormat {
+    return this.getRecordFormats(record)[0]
+  }
+
+  secondToLastFormat(record: CatalogRecord): FileFormat[] {
+    const formats = this.getRecordFormats(record)
+    return formats.slice(formats.length - 2)
```

Comment:
I believe there is a logic error here, to get the content of an array _without_ the first element you need to do `formats.slice(1)`; the associated test seems to be passing by coincidence I think :slightly_smiling_face: